--- a/as/dwarf2dbg.c	2019-01-16 21:54:53.962260186 -0800
+++ b/as/dwarf2dbg.c	2019-01-16 21:56:26.559824777 -0800
@@ -56,7 +56,7 @@
 #endif
 
 #include "dwarf2dbg.h"
-#include <filenames.h>
+#include "filenames.h" // Patch Jun 21, 2018 - Alexander Barker
 
 #ifdef HAVE_DOS_BASED_FILE_SYSTEM
 /* We need to decide which character to use as a directory separator.
@@ -1577,6 +1577,9 @@
     }
   else
     {
+      expr.X_add_number = 0;
+      emit_expr (&expr, 0);
+      sizeof_offset = 0;
       as_fatal (_("internal error: unknown dwarf2 format"));
     }
 
