--- a/ld64/src/ld/Snapshot.cpp	2012-09-24 18:53:30.000000000 -0700
+++ b/ld64/src/ld/Snapshot.cpp	2018-06-28 03:55:12.095982620 -0700
@@ -16,14 +16,18 @@
 #include <libgen.h>
 #include <time.h>
 #include <Block.h>
+#include <sys/param.h> // Patch Jun 28, 2018 - Alexander Barker
 
 #include "Snapshot.h"
 #include "Options.h"
 
-#include "compile_stubs.h"
+//#include "compile_stubs.h" // Patch Jun 26, 2018 - Alexander Barker
+#define ARG_MAX       131072
 
 //#define STORE_PID_IN_SNAPSHOT 1
 
+static const char *compile_stubs ="Empty"; // Patch Jun 26, 2018 - Alexander Barker
+
 // Well known snapshot file/directory names. These appear in the root of the snapshot.
 // They are collected together here to make managing the namespace easier.
 static const char *frameworksString         = "frameworks";         // directory containing framework stubs (mach-o files)
