--- a/libemulated/strmode.c	2018-06-27 23:43:59.275087423 -0700
+++ b/libemulated/strmode.c	2018-06-27 23:22:25.659483233 -0700
@@ -26,12 +26,13 @@
  * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
  * SUCH DAMAGE.
  */
-
-#if defined(LIBC_SCCS) && !defined(lint)
-static char sccsid[] = "@(#)strmode.c	8.3 (Berkeley) 8/15/94";
-#endif /* LIBC_SCCS and not lint */
-#include <sys/cdefs.h>
-__FBSDID("$FreeBSD: src/lib/libc/string/strmode.c,v 1.8 2009/04/14 11:39:56 trasz Exp $");
+// Begin Patch Jun 27, 2018 - Alexander Barker
+//#if defined(LIBC_SCCS) && !defined(lint)
+//static char sccsid[] = "@(#)strmode.c	8.3 (Berkeley) 8/15/94";
+//#endif /* LIBC_SCCS and not lint */
+//#include <sys/cdefs.h>
+//__FBSDID("$FreeBSD: src/lib/libc/string/strmode.c,v 1.8 2009/04/14 11:39:56 trasz Exp $");
+// End Patch
 
 #include <sys/types.h>
 #include <sys/stat.h>
