--- a/libstuff/llvm.c	2019-03-19 19:16:34.535294673 -0700
+++ b/libstuff/llvm.c	2019-03-19 20:38:07.157415028 -0700
@@ -13,7 +13,7 @@
  * plan to include it (along with the current libLTO APIs) in a generic
  * libLLVM.dylib.
  */
-#define LIB_LLVM "libLTO.dylib"
+#define LIB_LLVM "libLTO.so" // Patch Dec 28, 2014 - Alex Barker
 
 static int tried_to_load_llvm = 0;
 static void *llvm_handle = NULL;
@@ -66,10 +66,7 @@
 	    if(llvm_handle == NULL){
 		free(llvm_path);
 		llvm_path = NULL;
-		llvm_handle = dlopen("/Applications/Xcode.app/Contents/"
-				     "Developer/Toolchains/XcodeDefault."
-				     "xctoolchain/usr/lib/" LIB_LLVM,
-				     RTLD_NOW);
+		llvm_handle = dlopen("/usr/lib/llvm/" LIB_LLVM, RTLD_NOW); // Patch Dec 28, 2014 - Alex Barker
 	    }
 	    if(llvm_handle == NULL)
 		return(0);
